server/src/main/scala/za/co/absa/atum/server/api/controller/PartitioningController.scala

```diff
@@ -32,6 +32,12 @@ trait PartitioningController {
     partitioningSubmitDTO: PartitioningSubmitV2DTO
   ): IO[ErrorResponse, (SingleSuccessResponse[PartitioningWithIdDTO], String)]
 
+  def patchPartitioningParentV2(
+     partitioningId: Long,
+     parentPartitioningID: Long,
+     byUser: String
+ ): IO[ErrorResponse, SingleSuccessResponse[ParentPatchV2DTO]]
```

Comment:
I am not sure about the return value. Usually PATCH returns updated resource. In our case though we edit/establish relationship in the db. I wonder if instead we should return partitioning with all its parent partitionings or return 200 Ok or 204 No Content.

Reply:
I am for 200 Ok without any data, or 204 No Content.

* the Id makes no sense
* the partitioning doesn't give much value
* and partitioning with all parents while occasionally useful could be taxing on the DB - better to request explicitly for those rare occasions